### PR TITLE
Utilize DI for payload converter

### LINKED ISSUE #335: Use DI for ODataPayloadValueConverter
Rather than have users to implement their own model builder to register their `ODataPayloadValueConverter`. Use DI is much more flexible.

### PR DESCRIPTION
**Pull request by chinadragon0515**
Fix #335 , user can register his own payload converter.

### CODE REVIEW

File: src/Microsoft.Restier.WebApi/HttpConfigurationExtensions.cs [hunk 1]
```diff
@@ -47,15 +47,22 @@ public static class HttpConfigurationExtensions
...
             {
-                services.AddScoped<RestierQueryExecutorOptions>()
+                services = services.AddScoped<RestierQueryExecutorOptions>()
                     .ChainPrevious<IQueryExecutor, RestierQueryExecutor>();
+
+                // This means there is no customized converted registered
+                if (!services.HasService<ODataPayloadValueConverter>())
+                {
+                    services.AddSingleton<ODataPayloadValueConverter, RestierPayloadValueConverter>();
+                }
```
Comment (rayao) [inline src/Microsoft.Restier.WebApi/HttpConfigurationExtensions.cs]: - Could be replaced with TryAddScoped
- converted -> converter

File: src/Microsoft.Restier.WebApi/HttpConfigurationExtensions.cs [hunk 2]
```diff
@@ -65,6 +72,11 @@ public static class HttpConfigurationExtensions
 
                 var route = config.MapODataServiceRoute(
                     routeName, routePrefix, model, new DefaultODataPathHandler(), conventions, batchHandler);
+
+                // Customized converter should be added in ConfigureApi as service
+                var converter = api.Context.GetApiService<ODataPayloadValueConverter>();
```
Comment (rayao) [inline src/Microsoft.Restier.WebApi/HttpConfigurationExtensions.cs]: If the converter will always be retrieved from IEdmModel later, we can simply add
`if (converter == null) converter = new RestierPalyloadValueConverter();`
and omit the default registration.

File: src/Microsoft.Restier.WebApi/HttpConfigurationExtensions.cs (hunk 1 above)
Comment (lewischeng-ms) [inline src/Microsoft.Restier.WebApi/HttpConfigurationExtensions.cs]: TryAddScoped...Awesome!

File: src/Microsoft.Restier.WebApi/HttpConfigurationExtensions.cs (hunk 2 above)
Comment (lewischeng-ms) [inline src/Microsoft.Restier.WebApi/HttpConfigurationExtensions.cs]: +1

**Comment by lewischeng-ms** [comment]
LGTM

File: src/Microsoft.Restier.WebApi/HttpConfigurationExtensions.cs (hunk 1 above)
Comment (chinadragon0515) [inline src/Microsoft.Restier.WebApi/HttpConfigurationExtensions.cs]: Will omit this registration.

File: src/Microsoft.Restier.WebApi/HttpConfigurationExtensions.cs (hunk 2 above)
Comment (chinadragon0515) [inline src/Microsoft.Restier.WebApi/HttpConfigurationExtensions.cs]: Yeah, this also help to resolve an issue of some our test cases which calls Api configuration before calling map route, and cause null reference exception (no exception in normal case which calls map route first)
